libs/vkd3d/command.c

```diff
@@ -12504,10 +12518,19 @@ static void STDMETHODCALLTYPE d3d12_command_list_BuildRaytracingAccelerationStru
         return;
     }
 
+    /* Do not batch TLAS and BLAS builds into the same command, since doing so
+     * is disallowed if there are data dependencies between the builds. This
+     * happens in Cyberpunk 2077, which does not emit appropriate UAV barriers. */
+    if (rtas_batch->build_info_count && rtas_batch->build_type != desc->Inputs.Type)
+        d3d12_command_list_flush_rtas_batch(list);
```

Comment:
If there are no appropriate barriers, should be add them here? Seems like app is doing:
BOTTOM -> BOTTOM -> BOTTOM -> no barrier -> TOP -> TOP -> TOP?

Reply:
Can do, I omitted them because things just happen to work™ but then again this is only expected to happen once/frame anyway, so it won't hurt performance.

Reply:
Might randomly fix issues on NV perhaps.